File: 0198-house-robber/0198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        
        int n = nums.length;

        if (n == 1)
            return nums[0];

        int prev = nums[0];
        int secondPrev = 0;

        int loot = 0;
        for (int i = 1; i < n; i++) {
            loot = Math.max(nums[i] + secondPrev, prev);
            secondPrev = prev;
            prev = loot;
        }
        return loot;
    }
}

File: 0198-house-robber/0198-house-robber.ts
function rob(nums: number[]): number {
    const n = nums.length;

    if (n === 1) {
        return nums[0];
    }

    let prev = nums[0];
    let secondPrev = 0;

    let loot = 0;
    for (let i = 1; i < n; i++) {
        loot = Math.max(nums[i] + secondPrev, prev);
        secondPrev = prev;
        prev = loot;
    }

    return loot;
}
